### ISSUE TITLE
AFK channel support by allowing a channel to be Deafened

### ISSUE THREAD

**Issue by Tarun80** [mumble-voip/mumble#1257]
I briefly touched on this over a year ago on the [forums](http://sourceforge.net/apps/phpbb/mumble/viewtopic.php?f=2&t=1011).

Previously I suggested:
> I'd also like to see the ability to have a true AFK channel where ALL audio messages become suppressed. Currently you can deny Speak, but people in the AFK channel can still hear when people connect, disconnect, join channel, etc. If possible, I'd like to see it where people joining the AFK channel are also automatically deafened as well based upon how you set up the permissions. So they would be able to be muted and deafened automatically when joining the AFK channel.

I, along with many other users of the server I host would like to see this implemented. It would also be very useful when moving users who have been idle for an extended amount of time. I move users who have been idle for an hour or so and I would very much like to see them automatically be deafened when placed into the AFK channel. Mainly because you never know if they use speakers or not and I see this as a polite courtesy. By allowing the AFK channel to be deafened this would stop the connect/disconnect and join/leave channel messages/audio cues.

Speaking of idle time. It would be nice if we could also have an option to automatically move users to the AFK channel after a designated amount of idle time. These could be options in the ini to allow hosts to configure this to their liking. Ventrilo also has this feature supported.

I can also expand on the details of this if necessary. But basically just being able to allow a channel to be Deafened would be very nice. I believe Teamspeak can do this, and I know that [Ventrilo](http://www.instantventrilo.com/static/en/img/support/new/ventrilo-setup-channel.jpg) can do this. TTS events, audio cues, and so forth are all silent for users in the AFK channel.

Thanks Mumble team.

**Comment by Tarun80** [comment]
@Kissaki, @mkrautz, can this please be reopened and proper support for an AFK channel be implemented? All we really need is a Deny privilege for Listen or something simple, just like how you can deny Speak.

**Comment by Tarun80** [comment]
Any update on this? Even Discord has AFK channel support.

**Comment by Kissaki** [comment]
> All we really need is a Deny privilege for Listen or something simple, just like how you can deny Speak.

ACL already has a lot of permissions, An additional will make it even more complex.

I still don’t see the use. IMO the client should be in control of whether he mutes himself or not when going afk, not the server. If you watch a movie, read a book, whatever, but want to hear if someone joins that would not be possible with such an AFK channel.

We also have no sane concept for idle recognition either. And if a user is determined idle client side, how does the server know, and what should he do? A simple Mumble inactivity -> move can already be done with Ice scripts.

Anyway, I see what you see, and it seems nobody is working on this right now.

**Comment by wrapper** [comment]
+1 2020 would still be a good feature

**Comment by Krzmbrzl** [comment]
@wrapper it'd be better if you +1'ed via emoji on the initial post as this can be filtered for ;)

**Comment by dstensnes** [comment]
Cannot [Mumo](https://wiki.mumble.info/wiki/Mumo) be used for this?

**Comment by ZeroAbility** [comment]
> Cannot [Mumo](https://wiki.mumble.info/wiki/Mumo) be used for this?

Yes:

https://github.com/Natenom/mumblemoderator-module-collection/tree/master/deaftoafk

Has there been any features implemented that addressed this? Most OS's have the ability to mute audio by application in their mixer software.

**Comment by Hartmnt** [comment]
> Has there been any features implemented that addressed this? Most OS's have the ability to mute audio by application in their mixer software.

No, the idea of OP is that, since deafening mutes all notification sounds, a channel should be able to override the user deafened status (at least one way). This would be like the "Speak" ACL, but for listening and receiving notifications.
Whether or not this is actually needed in Mumble or not is up for debate though I guess.

**Comment by ZeroAbility** [comment]
It should suffice to just make a channel that doesn't inherit permissions from the parent. No idea if `@admin` members bypass ACL security, but an admin should only enter that room to possibly remove a user from it or the server. Are you saying the ACL does not provide the function as advertised?

**Comment by Hartmnt** [comment]
> Are you saying the ACL does not provide the function as advertised?

Simply put: There is no ACL that behaves like "Speak" but for the user deafened status. "Mute/Deafen" does not apply here as it requires admin interaction.
To be consistent with the naming of other ACLs the new, hypothetical ACL for this would be named "Hear"
